bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -311,13 +322,27 @@ public synchronized NeoHubReturnResult toNeoHubSendChannelValue(String commandSt
             }
 
             if (systemData == null) {
-                logger.warn(MSG_FMT_SYSTEM_POLL_ERR, "failed to create system data response");
+                logger.warn(MSG_FMT_SYSTEM_POLL_ERR, getThing().getUID(), "failed to create system data response");
                 return null;
             }
 
+            String physicalFirmware = systemData.getFirmwareVersion();
+            if (physicalFirmware != null) {
+                String thingFirmware = getThing().getProperties().get(PROPERTY_FIRMWARE_VERSION);
+                if (!physicalFirmware.equals(thingFirmware)) {
+                    // update the thing firmware property to the hub's physical firmware value
+                    getThing().setProperty(PROPERTY_FIRMWARE_VERSION, physicalFirmware);
+                    // log changes, but only if the prior property value was neither null nor empty
+                    if ((thingFirmware != null) && (thingFirmware.length() > 0)) {
+                        logger.info("hub '{}' has updated its firmware from '{}' to '{}'", getThing().getUID(),
+                                thingFirmware, physicalFirmware);
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug.

Same for below.

Reply:
> Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded

@fwolter thank you for the review. The Bridge firmware gets updated about 2..3 times per year (which seems quite 'sparse' to me). There would be no point in logging at DEBUG level, since the coincidence of having DEBUG turned on at the same instant as a firmware update might occur, is not likely. I would prefer to keep it at INFO level (but if you insist on it not being INFO then I would delete this logger line entirely). => Please advise.

> Same for below

I guess you are referring to line 392 ?? -- If so the argument is similar to above. The Heatmiser RF mesh network is very solid and generally devices never drop out of the network at all. However 'just in case' it is useful for the user to know if a device did exceptionally drop out. Again, the coincidence of having DEBUG turned on at the same instant as a dropout might occur, is not likely. => Ditto, please advise.

Reply:
With "sparsely" I meant not in time, but the number of occurrences in the code. Although it doesn't comply with the coding guidelines, I won't insist on changing it.

Reply:
@fwolter I could delete the logging in the case you mentioned first. But in the the case you mentioned second, I would like to keep that on INFO. => Is that Ok with you?

Reply:
Ok. I made a new commit with the changes proposed above.